tests/api2/test_attachment_querying.py

```diff
@@ -0,0 +1,34 @@
+#!/usr/bin/env python3
+
+import os
+import sys
+from pytest_dependency import depends
+
+sys.path.append(os.getcwd())
+
+from middlewared.test.integration.assets.nfs import nfs_share
+from middlewared.test.integration.assets.pool import dataset
+from middlewared.test.integration.utils import call, client
+
+
+PARENT_DATASET = 'test_parent'
+CHILD_DATASET = f'{PARENT_DATASET}/child_dataset'
+
+
+def test_attachment_with_child_path(request):
+    depends(request, ['pool_04'], scope='session')
+    with dataset(PARENT_DATASET) as parent_dataset:
+        parent_path = f'/mnt/{parent_dataset}'
+        assert call('pool.dataset.attachments_with_path', parent_path) == []
+
+        with nfs_share(parent_dataset):
+            attachments = call('pool.dataset.attachments_with_path', parent_path)
+            assert len(attachments) > 0, attachments
+            assert attachments[0]['type'] == 'NFS Share', attachments
+
+            with dataset(CHILD_DATASET) as child_dataset:
+                child_path = f'/mnt/{child_dataset}'
+                call('sharing.nfs.create', {'path': child_path})
```

Comment:
Why not also use a context manager that will remove the test share upon exiting?

Reply:
Ah sorry, missed this - this is not intended